From a3ffbe7924b65943776d268bab1b2e5a9ca8b8c1 Mon Sep 17 00:00:00 2001
From: AsavarTzeth <asavartzeth@gmail.com>
Date: Sun, 12 Aug 2018 19:45:19 +0200
Subject: [PATCH 1/3] Fix issues with setup.py installation paths

There are some minor issues with where CHIRP installs files, such as
locales.

- On Linux, locales should be installed under $prefix/share directly,
  and not in any application specific sub-directory.

- For all of this to work in a Flatpak environment the /app prefix has
  to be supported. Adding "/app/share" as another share location
  candidate currently solves this.
---
 chirp/platform.py | 1 +
 setup.py          | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/chirp/platform.py b/chirp/platform.py
index 9b2922b..485874d 100644
--- a/chirp/platform.py
+++ b/chirp/platform.py
@@ -249,6 +249,7 @@ class Platform:
         share_candidates = [
             os.path.join(execpath, "share"),
             os.path.join(sys.prefix, "share"),
+            "/app/share",
             "/usr/local/share",
             "/usr/share",
         ]
diff --git a/setup.py b/setup.py
index 837d4e6..9f5df43 100644
--- a/setup.py
+++ b/setup.py
@@ -115,7 +115,7 @@ def default_build():
 
     locale_files = []
     for f in _locale_files:
-        locale_files.append(("share/chirp/%s" % os.path.dirname(f), [f]))
+        locale_files.append(("share/%s" % os.path.dirname(f), [f]))
 
     print "LOC: %s" % str(locale_files)
 
-- 
2.18.0

